set_alias.py

```diff
@@ -38,38 +50,90 @@ def run(self):
         on-call user group in slack to be the on-call users
         """
 
-        slack_users = self.slack_users_by_email(self.on_call_email_addresses)
+        today = datetime.now(dateutil.tz.gettz(settings.TIMEZONE)).replace(hour=0, minute=0)
+
+        weekday = today.weekday()
+        oncalls = self.get_on_call_engs(weekday)
+        slack_users = self.slack_users_by_email(oncalls)
         if not slack_users:
-            log.warning('No Slack users found for email addresses: %s', ','.join(self.on_call_email_addresses))
+            log.warning('No Slack users found for email addresses: {0}'.format(','.join(oncalls)))
             return
 
-        slack_user_ids = [u['id'] for u in slack_users]
+        # self.backfill_calendar_events(today, settings.DAYS_AHEAD)
 
+        if settings.GOOGLE_CAL_ID:
+            # set on-call calendar event for the future
+            the_future = today + timedelta(days=self.days_ahead)
+            weekday = the_future.weekday()
+            oncalls = self.get_on_call_engs(weekday)
+            self.set_calendar_event(oncalls, the_future)
+
+        # update slack oncall
+        slack_user_ids = [u['id'] for u in slack_users]
         if set(slack_user_ids) == set(self.slack_user_group['users']):
-            log.info('User group %s already set to %s', self.slack_user_group_handle, slack_user_ids)
+            log.info(f'User group {self.slack_user_group_handle} already set to {slack_user_ids}')
             return
-
         self.update_on_call(slack_users)
+
         log.info('Job Complete')
 
-    def _make_request(self, url, body=None, headers={}):
-        #  TODO: now that we have serverless we can re-write this archaic dodo with requests
-        headers['Authorization'] = f'Bearer {self.slack_token}'
-        req = Request(url, body, headers)
-        log.info('Making request to %s', url)
+    def _make_request(self, url):
+        response = requests.get(url, headers={'Authorization': f'Bearer {self.slack_token}'})
+        return response.json()
 
-        try:
-            response = urlopen(req)
-            body = response.read()
-            try:
-                return json.loads(body)
-            except ValueError:
-                return body
+    def backfill_calendar_events(self, start_date, num_days):
+        """
+        Do not run unless you are prepared for calendar spam. Do not start on a sunday since this assumes Saturday
```

Comment:
why not pre-calculate weekend rotation for forever with re-occurring events and then just validate its correct/ update if someone gets added

Reply:
because i didn't want people planning vacations > 45 days in advance to have the incorrect info.  also, it's nice to have that guarantee that the event won't change after creation so that people can swap days if needed